stock_request_purchase/models/purchase_order_line.py

```diff
@@ -10,6 +11,16 @@ class PurchaseOrderLine(models.Model):
     stock_request_ids = fields.Many2many(comodel_name='stock.request',
                                          string='Stock Requests', copy=False)
 
+    @api.constrains('stock_request_ids')
+    def _check_purchase_company_constrains(self):
+        for rec in self:
+            for request in rec.stock_request_ids:
```

Comment:
You can replace the two for loops by a single loop: `for request in self.mapped('stock_request_ids'):`.

Reply:
But then you loose the ability to do the check 'if request.company_id != rec.company_id', because you do not have rec.